# Missing includes
diff -u a/dios/CMakeLists.txt b/dios/CMakeLists.txt
--- a/dios/CMakeLists.txt	2019-11-01 15:57:40.145739566 +0100
+++ b/dios/CMakeLists.txt	2019-11-01 15:57:40.145739566 +0100
@@ -10,7 +10,7 @@
       include/netinet/*.h include/arpa/*.h
       libcxxabi/include/*
       libcxxabi/src/*.h libcxxabi/src/*.hpp libcxxabi/src/*.ipp
-      libcxx/include/* libcxx/ext/* libcxx/include/experimental/*
+      libcxx/include/* libcxx/include/ext/* libcxx/include/experimental/*
       libcxx/src/*.h
       libcxx/include/support/xlocale/*.h libcxx/src/support/atomic_support.h
       libcxx/include/support/divine/*.h

# Fix compilation of shoop
diff -u a/dios/CMakeLists.txt b/dios/CMakeLists.txt
--- a/tools/shoop.cpp	2020-01-21 10:50:46.670265138 +0100
+++ b/tools/shoop.cpp	2020-01-21 13:09:18.709691652 +0100
@@ -322,9 +322,12 @@
 
     // Now let's run Divine
     auto bc = std::make_shared< divine::mc::BitCode >( std::move( tested_mod ), ctx );
-    bc->dios_config( "static" );
 
-    bc->symbolic( true );
+    divine::mc::BCOptions bco;
+    bco.dios_config = "static";
+    bco.symbolic = true;
+
+    bc->set_options( bco );
     bc->solver( "stp" );
     bc->init();
 
@@ -356,7 +359,7 @@
         Stepper step;
         step._stop_on_error = true;
         step._stop_on_accept = true;
-        step._ff_components = divine::dbg::Component::Kernel;
+        step._ff_components = divine::dbg::component::kernel;
         step.run( dbg, Stepper::Quiet );
 
         log->backtrace( dbg, 10 ); // How detailed we want the log to be?
