# Testsuite
diff --git a/test/lib/testcase b/test/lib/testcase
--- a/test/lib/testcase
+++ b/test/lib/testcase
@@ -29,7 +29,7 @@
 
 draw() { divine draw --render cat "$@"; }
 
-PATH="$PWD/lib:$TOOLS:$TOOLS/../llvm/bin:$PATH"
+PATH="$PWD/lib:$TOOLS:$TOOLS/../llvm/bin:$TOOLS/../clang/bin:$PATH"
 extracheck=:
 export cesmiext=.so
 test "$WIN32" = 1 && cesmiext=.dll

diff --git a/test/lib/verify b/test/lib/verify
--- a/test/lib/verify
+++ b/test/lib/verify
@@ -84,7 +84,7 @@ 
     addcmd load $dir/$f $f
     for dep in $deps; do
         if ! echo $dep | grep -q '^/'; then dep=$dir/$dep; fi
-        addcmd load $dep $(echo $dep | sed -re "s,$dir/*,,")
+        addcmd load $dep $(basename $dep)
     done
 done
 
# Missing includes
diff -u a/dios/CMakeLists.txt b/dios/CMakeLists.txt
--- a/dios/CMakeLists.txt	2019-11-01 15:57:40.145739566 +0100
+++ b/dios/CMakeLists.txt	2019-11-01 15:57:40.145739566 +0100
@@ -10,7 +10,7 @@
       include/netinet/*.h include/arpa/*.h
       libcxxabi/include/*
       libcxxabi/src/*.h libcxxabi/src/*.hpp libcxxabi/src/*.ipp
-      libcxx/include/* libcxx/ext/* libcxx/include/experimental/*
+      libcxx/include/* libcxx/include/ext/* libcxx/include/experimental/*
       libcxx/src/*.h
       libcxx/include/support/xlocale/*.h libcxx/src/support/atomic_support.h
       libcxx/include/support/divine/*.h

# getline
diff -u a/dios/libc/stdio/getline.c b/dios/libc/stdio/getline.c
--- a/dios/libc/stdio/getline.c	2019-11-15 14:29:47.130277649 +0100
+++ b/dios/libc/stdio/getline.c	2019-11-15 14:29:47.133610968 +0100
@@ -23,7 +23,7 @@
     if ( !(*lineptr) || *n <= 1 )
     {
         *lineptr = realloc( *lineptr, INIT_SIZE );
-        if ( !lineptr )
+        if ( !(*lineptr) )
             return -1;
         *n = INIT_SIZE;
     }

