diff -Nur a/Makefile b/Makefile
--- a/Makefile	2019-07-16 02:14:03.000000000 +0200
+++ b/Makefile	2019-07-26 11:29:25.679836238 +0200
@@ -14,7 +14,7 @@
 RELEASE_BUILD_TYPE ?= RelWithDebInfo
 
 MAKEFLAGS ?= --no-print-directory
-OBJ ?= $(PWD)/_build.
+OBJ ?= $(CURDIR)/_build.
 BENCH_NAME ?= $(LOGNAME)
 EXTRA != if test "$(GENERATOR)" = Ninja && test -n "$(VERBOSE)"; then echo -v -d explain; fi; \
          if test -n "$(JOBS)"; then echo -j $(JOBS); fi
@@ -22,7 +22,7 @@
 TOOLDIR = $(OBJ)toolchain
 CLANG = $(TOOLDIR)/clang/
 RTBIN = $(TOOLDIR)/dios
-RTSRC = $(PWD)/dios
+RTSRC = $(CURDIR)/dios
 
 LIBUNWIND_LDIR = $(RTBIN)/libunwind/src
 CXX_LDIR = $(TOOLDIR)/lib
@@ -94,7 +94,7 @@
 	$(CMAKE) -E copy_if_different $(OBJ)$(FLAVOUR)/config.tmp $(OBJ)$(FLAVOUR)/config.vars
 	if ! test -e $(OBJ)$(FLAVOUR)/config.done || test -n "$(FORCE_CMAKE)"; then \
 	  chmod +x test/divine; \
-	  cd $(OBJ)$(FLAVOUR) && $(CMAKE) $(PWD) $(CMAKE_EXTRA) -G "$(GENERATOR)" && \
+	  cd $(OBJ)$(FLAVOUR) && $(CMAKE) $(CURDIR) $(CMAKE_EXTRA) -G "$(GENERATOR)" && \
 	  touch $(OBJ)$(FLAVOUR)/config.done; fi
 
 build: config
diff -Nur a/stp/lib/Interface/CMakeLists.txt b/stp/lib/Interface/CMakeLists.txt
--- a/stp/lib/Interface/CMakeLists.txt	2019-07-16 02:14:41.000000000 +0200
+++ b/stp/lib/Interface/CMakeLists.txt	2019-07-26 11:29:25.703169571 +0200
@@ -40,7 +40,7 @@
         add_custom_command(TARGET CopyPublicHeaders PRE_BUILD
                            COMMAND ${CMAKE_COMMAND} -E make_directory ${HEADER_DEST}
                            COMMAND ${CMAKE_COMMAND} -E echo "LEGACY: Copying ${public_header} to ${HEADER_DEST}"
-                           COMMAND ${CMAKE_COMMAND} -E copy_if_different "${PROJECT_SOURCE_DIR}/include/stp/${public_header}" "${HEADER_DEST}/${public_header}"
+                           COMMAND ${CMAKE_COMMAND} -E copy_if_different "${PROJECT_SOURCE_DIR}/stp/include/stp/${public_header}" "${HEADER_DEST}/${public_header}"
                           )
     endforeach()
 endif()

diff -Nur a/tools/CMakeLists.txt b/tools/CMakeLists.txt
--- a/tools/CMakeLists.txt	2019-08-01 23:51:33.000000000 +0200
+++ b/tools/CMakeLists.txt	2019-08-06 16:10:49.335477692 +0200
@@ -66,7 +66,7 @@
   set( INSTALL_BIN_DEST bin )
 endif( WIN32 )
 
+install( TARGETS divine divcc dioscc runtime-cc runtime-ld lart divcheck
-install( TARGETS divine
          DESTINATION ${INSTALL_BIN_DEST} COMPONENT console_tools )
 
 # NB. Must be here (and not in the toplevel cmakelists) to ensure correct
