diff -rN -u a/divine/cc/driver.cpp b/divine/cc/driver.cpp
--- a/divine/cc/driver.cpp	2019-11-01 15:57:40.145739566 +0100
+++ b/divine/cc/driver.cpp	2019-11-01 15:57:40.145739566 +0100
@@ -213,7 +213,7 @@
         compiler.allowIncludePath( "." );  // TODO: necessary?
         compiler.allowIncludePath( brick::string::dirname( lib ) );
 
-        dirs.insert( dirs.begin(), "/dios/lib" );
+        dirs.insert( dirs.begin(), "/opt/divine/include/dios/lib" );
         for ( auto p : dirs )
             for ( auto suf : suffixes )
                 for ( auto pref : { "lib", "" } )
@@ -227,7 +229,7 @@
 
     std::string Driver::find_object( std::string name )
     {
-        return joinPath( "/dios/lib", name + ".bc"s );
+        return joinPath( "/opt/divine/include/dios/lib", name + ".bc"s );
     }
 
     Driver::ModulePtr Driver::load_object( std::string path )

diff -rN -u a/dbg/info.cpp b/divine/dbg/info.cpp
--- a/divine/dbg/info.cpp	2019-11-01 15:57:40.145739566 +0100
+++ b/divine/dbg/info.cpp	2019-11-01 15:57:40.145739566 +0100
@@ -60,28 +60,28 @@
 
     if ( comp & Component::LibC )
     {
-        if ( startsWith( file, "/dios/libc/" ) ||
-             startsWith( file, "/dios/arch/" ) ||
-             startsWith( file, "/dios/include/sys/" ) )
+        if ( startsWith( file, "/opt/divine/include/dios/libc/" ) ||
+             startsWith( file, "/opt/divine/include/dios/arch/" ) ||
+             startsWith( file, "/opt/divine/include/dios/include/sys/" ) )
             return true;
-        if ( startsWith( file, "/dios/include/" ) && file.substr( 14 ).find( '/' ) == std::string::npos )
+        if ( startsWith( file, "/opt/divine/include/dios/include/" ) && file.substr( 33 ).find( '/' ) == std::string::npos )
             return true;
     }
 
     if ( comp & Component::LibCxx )
-        if ( startsWith( file, "/dios/libcxx" ) )
+        if ( startsWith( file, "/opt/divine/include/dios/libcxx" ) )
             return true;
 
     if ( comp & Component::LibRst )
-        if ( startsWith( file, "/dios/rst/" ) ||
-             startsWith( file, "/dios/include/rst/" ) )
+        if ( startsWith( file, "/opt/divine/include/dios/rst/" ) ||
+             startsWith( file, "/opt/divine/include/dios/include/rst/" ) )
             return true;
 
     if ( comp & Component::DiOS )
-        if ( startsWith( file, "/dios/sys/" ) || startsWith( file, "/dios/vfs/" ) )
+        if ( startsWith( file, "/opt/divine/include/dios/sys/" ) || startsWith( file, "/opt/divine/include/dios/vfs/" ) )
             return true;
 
-    if ( comp & Component::Program && !startsWith( file, "/dios/" ) )
+    if ( comp & Component::Program && !startsWith( file, "/opt/divine/include/dios/" ) )
         return true;
 
     return false;

diff -rN -u a/divine/rt/dios-cc.cpp b/divine/rt/dios-cc.cpp
--- a/divine/rt/dios-cc.cpp	2019-11-01 15:57:40.145739566 +0100
+++ b/divine/rt/dios-cc.cpp	2019-11-01 15:57:40.145739566 +0100
@@ -30,10 +30,10 @@
     using namespace brick::fs;
 
     paths.insert( paths.end(),
-                 { "-isystem", "/dios/libcxx/include"
-                 , "-isystem", "/dios/libcxxabi/include"
-                 , "-isystem", "/dios/libunwind/include"
-                 , "-isystem", "/dios/include" } );
+                 { "-isystem", "/opt/divine/include/dios/libcxx/include"
+                 , "-isystem", "/opt/divine/include/dios/libcxxabi/include"
+                 , "-isystem", "/opt/divine/include/dios/libunwind/include"
+                 , "-isystem", "/opt/divine/include/dios/include" } );
 }
 
 void add_dios_defines( std::vector< std::string >& flags )

diff -rN -u a/divine/rt/runtime.cpp b/divine/rt/runtime.cpp
--- a/divine/rt/runtime.cpp	2019-11-01 15:57:40.145739566 +0100
+++ b/divine/rt/runtime.cpp	2019-11-01 15:57:40.145739566 +0100
@@ -33,11 +33,11 @@
         using brick::string::endsWith;
 
         if ( n == "libcxx.a" ) n = "libc++.a";
-        if ( n == "libcxxabi.a" ) n = "libc++abi.a";
+        if ( n == "libcxxabi.a") n = "libc++abi.a";
         if ( endsWith( n, ".bc" ) || endsWith( n, ".a" ) )
-            return "/dios/lib/" + n;
+            return "/opt/divine/include/dios/lib/" + n;
         else
-            return "/dios/" + n;
+            return "/opt/divine/include/dios/" + n;
     }
 
     void each( std::function< void( std::string, std::string_view ) > yield )

diff -rN -u a/tools/runtime-cc.cpp b/tools/runtime-cc.cpp
--- a/tools/runtime-cc.cpp	2019-11-01 15:57:40.145739566 +0100
+++ b/tools/runtime-cc.cpp	2019-11-01 15:57:40.145739566 +0100
@@ -48,6 +48,7 @@
                 n = p.substr( srcDir.size() );
             if ( string::startsWith( p, binDir ) )
                 n = p.substr( binDir.size() );
+            n = "/opt/divine/include" + n;
             TRACE( "rewrite", p, "to", n );
             return n;
         } );
